libraries/Microsoft.Bot.Builder.Ai/StartupTranslationAndLocale.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Microsoft.Bot.Schema;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.Ai
+{
+    /*
+     * StartupTranslationAndLocale base class for delegate functions used to change language for the translator middleware
+     * 
+     */
+    public abstract class StartupTranslationAndLocale
+    {
+        private void SetLanguage(IBotContext context, string language) => context.State.User[@"Microsoft.API.translateTo"] =  language ;
+        private void SetLocale(IBotContext context, string locale) => context.State.User[@"LocaleConverterMiddleware.fromLocale"] = locale;
+
+        protected abstract   bool IsSupportedLanguage(string language);
+        protected virtual async Task<bool> SetActiveLanguage(IBotContext context)
+        {
+            bool changeLang = true;//logic implemented by developper to make a signal for language changing 
+            //use a specific message from user to change language
+            if (changeLang)
+            {
+                var newLang = ""; //extracted by the user using user state 
+                if (!string.IsNullOrWhiteSpace(newLang)
```

Comment:
I may be missing the obvious, but I'm not clear how this code ever works. You're *always* creating newLang and setting to "". Then, in the very next line, you're checking to make sure it's not empty (which it is).

Reply:
This class is an abstract class (aka a sample to help the developper create a logic to change the language) so the newLang part will be changed in the implemented class by the developper and the logic of changing the language will be set by the developer